back/app/services/files/description_generator.rb

```diff
@@ -0,0 +1,199 @@
+      end
+    end
+
+    private
+
+    # Get configured locales for the application
+    #
+    # @return [Array<String>] Array of locale codes
+    def tenant_locales
+      @tenant_locales ||= AppConfiguration.instance.settings('core', 'locales')
+    end
+
+    # Generate descriptions using LLM for all configured locales
+    #
+    # @param content [String] The extracted file content
+    # @param locales [Array<String>] Array of locale codes
+    # @return [Hash] Hash with locale keys and description values
+    def generate_descriptions(file, locales)
+      schema = description_schema(locales)
+      chat = RubyLLM.chat(model: 'gpt-4.1').with_schema(schema)
+      prompt = build_prompt(file.name)
+
+      with_preprocessed_file_content(file) do |source|
+        chat.ask(prompt, with: source).content
+      end
+    end
+
+    # Yields the preprocessed file content to the block. If no preprocessing is needed,
+    # the original file content is passed. A block is used to ensure that any temporary
+    # files are properly closed and deleted after the block exits.
+    #
+    # @yield [source] The preprocessed file content
+    # @yieldparam [Tempfile, BaseFileUploader] source The preprocessed file content
+    def with_preprocessed_file_content(file, &)
+      preprocessed_file = preprocess_file(file)
+      source = preprocessed_file.is_a?(Tempfile) ? preprocessed_file.path : preprocessed_file.full_url
+      yield source
+    ensure
+      preprocessed_file.close! if preprocessed_file.is_a?(Tempfile)
+    end
+
+    # @return [Tempfile, BaseFileUploader]
+    def preprocess_file(file)
+      if file.text?
+        file.content
+      elsif file.image?
+        preprocess_image(file)
+      elsif previewable_as_pdf?(file)
+        preprocess_previewable_as_pdf(file)
+      else # rubocop:disable Lint/DuplicateBranch
+        file.content
+      end
+    end
+
+    # @return [Tempfile, BaseFileUploader]
+    def preprocess_previewable_as_pdf(file)
+      preview = file.preview
+
+      if preview.content.full_url.present?
+        preview.content
+      elsif preview.status == 'pending'
+        raise ApplicationJob::RetryInError.new('Preview is pending, try again later.', 30.seconds)
+      elsif preview.status == 'failed' && docx?(file)
+        docx_to_html(file)
+      else
+        # Just return the file content as is. It will most likely be rejected by the LLM
+        # unless support for additional file types is added. The alternative is to raise
+        # an error directly.
+        file.content
+      end
+    end
+
+    # @return [Tempfile] Tempfile with HTML content
+    def docx_to_html(file)
+      Tempfile.open do |local_file|
+        local_file.binmode.write(file.content.read)
+        doc = Docx::Document.open(local_file.path)
+
+        Tempfile
+          .new([file.name, '.html'])
+          .tap { |processed_file| processed_file.write(doc.to_html) }
+      end
+    end
```

Comment:
Not super sure whether we really need this, and the extra dependency that comes with it. Are you sure this is a meaningful improvement over the preview?